src/codegen/mod.rs

```diff
@@ -75,20 +80,69 @@ pub trait CodeGen<Tgt: Target> {
             println!("{}", String::from_utf8_lossy(&clang_proc.stderr));
         }
 
-        Ok(BuiltArtifact::new(binary_path, source_path, dirname, None))
+        Ok(BuiltArtifact::new(
+            binary_path,
+            source_path,
+            dirname,
+            bench_samples,
+        ))
+    }
+
+    /// Benchmark several times, returning the minimum of inner loop means.
+    //
+    //  This will first estimate a good number of inner loop iterations, then
+    //  build an executable which loops that number of times, returning the mean.
+    //  The final `result` computed is the minimum of the means after running
+    //  that executable `repeat` times.
+    fn time_impl_robustly(
+        &self,
+        print_code: bool,
+        repeat: Option<usize>,
+    ) -> Result<RobustTimingResult> {
+        let repeat = repeat.unwrap_or(10); // default: 10
+
+        // Collect a single rough sample.
+        let time_check_artifact = self.build(false, Some(1))?;
+        let rough_secs = time_check_artifact.measure_time()?;
+
+        // Choose a good number of iterations for benchmarks' inner loop.
+        let inner_iters = max(
+            MIN_SAMPLES,
+            (MIN_TRIAL_TIME_SECS / rough_secs).ceil() as usize,
+        );
```

Comment:
This can be, unfortunately, larger than can be represented with a signed literal in C. For example:
```bash
/var/folders/q8/lrhg0x1j0cqf125zmwc3pck80000gn/T/.tmpGkieXh/main.c:82:49: warning: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Wimplicitly-unsigned-literal]
  for (unsigned long bench_itr = 0; bench_itr < 18446744073709551615; ++bench_itr) {
```

I think this PR should, for now, adjust the generated code to silence that warning with `UL`. (Longer run, we should improve how we choose target benchmark duration.)

Reply:
@samkaufman What about using `unsigned long long` and `ULL` instead of `unsigned long` and `UL`? Since `usize` can be 8 bytes on a 64 bit target[^1], I think `unsigned long long`, which is also 8 bytes, would be more appropriate.

[^1]: https://doc.rust-lang.org/std/primitive.usize.html

Reply:
Since `Duration` only has division with `u32`, `f32`, or `f64`, I change the type from `usize` to `u32`. I just added `UL` suffix.